src/main/kotlin/org/malachite/estella/services/TaskService.kt

```diff
@@ -88,6 +97,22 @@ class TaskService(
                 )
             }
 
+    fun addResult(result: TaskResultRabbitDTO) {
+        val taskStage = taskStageService.getTaskStage(result.solverId)
+        val task = getTaskById(result.taskId).toTask()
+        val startTime = result.startTime?.let { if (it == "null") null else Timestamp.valueOf(it) } ?: let { null }
+        val endTime = result.endTime?.let { if (it == "null") null else Timestamp.valueOf(it) } ?: let { null }
```

Comment:
```suggestion
        val startTime = result.startTime
            ?.takeIf { it != "null" }
            ?.let { Timestamp.valueOf(it) }
        val endTime = result.endTime
            ?.takeIf { it != "null" }
            ?.let { Timestamp.valueOf(it) }
```
Niepotrzebny let { null } i imo można to opykać tak funkcyjnie, co by lepiej pokazywało co tu się dzieje, na szczęście w Kotlinie mamy podrabiany `Option#filter` w postaci `takeIf`